src/app.py

```diff
@@ -95,90 +103,94 @@ def query() -> Any:
         "status": "new",
         "rule_name": rules[-1].name,
         "rule_author": rules[-1].author,
-        "raw_yara": raw_yara,
+        "raw_yara": data.raw_yara,
         "submitted": int(time.time()),
-        "priority": req["priority"],
+        "priority": data.priority,
     }
 
-    if "taint" in req and req["taint"] is not None:
-        job_obj["taint"] = req["taint"]
+    if data.taint is not None:
+        job_obj["taint"] = data.taint
 
     redis.hmset("job:" + job_hash, job_obj)
     redis.rpush("queue-search", job_hash)
 
-    return jsonify({"query_hash": job_hash})
-
+    return QueryResponseSchema(query_hash=job_hash)
 
-@app.route("/api/matches/<hash>")
-def matches(hash: str) -> Response:
-    offset = int(request.args["offset"])
-    limit = int(request.args["limit"])
 
+@app.get("/api/matches/{hash}", response_model=MatchesSchema)
+def matches(
+    hash: str, offset: int = Query(...), limit: int = Query(...)
+) -> MatchesSchema:
     p = redis.pipeline(transaction=False)
     p.hgetall("job:" + hash)
     p.lrange("meta:" + hash, offset, offset + limit - 1)
     job, meta = p.execute()
-
-    return jsonify({"job": job, "matches": [json.loads(m) for m in meta]})
+    return MatchesSchema(job=job, matches=[json.loads(m) for m in meta])
+
+
+def get_job(job_id: str) -> JobSchema:
+    job = redis.hgetall(job_id)
+    return JobSchema(
+        id=job_id[4:],
+        status=job.get("status", "[error]"),
+        rule_name=job.get("rule_name", "[error]"),
+        rule_author=job.get("rule_author", None),
+        raw_yara=job.get("raw_yara", "[error]"),
+        submitted=job.get("submitted", 0),
+        priority=job.get("priority", "[error]"),
+        files_processed=job.get("files_processed", 0),
+        total_files=job.get("total_files", 0),
+    )
 
 
-@app.route("/api/job/<hash>")
-def job_info(hash: str) -> Response:
-    return jsonify(redis.hgetall("job:" + hash))
+@app.get("/api/job/{job_id}", response_model=JobSchema)
+def job_info(job_id: str) -> JobSchema:
+    return get_job(f"job:{job_id}")
```

Comment:
I guess this is to reduce API changes, but tbh responding with JSON that has only 1 key is pointless (just return value).

Reply:
Can you explain? I mean, JobSchema is an object with quite a few elements (including ID, status, etc)

Reply:
nvm I missread the code ._.